File: 프로그래머스/0/120842. 2차원으로 만들기/2차원으로 만들기.cs
using System;

public class Solution 
{
    public int[,] solution(int[] num_list, int n) 
    {
        int row = num_list.Length / n;
        int col = n;
        int[,] resultList = new int[row,col];
        
        int index = 0;
        int count = 0;
        for(int i=0; i<row; i++)
        {
            while(true)
            {
                resultList[i,count] = num_list[index];
                count++;
                index++;
                if(count == n)
                {
                    count = 0;
                    break;
                }
            }
        }
        return resultList;
    }
}

File: 프로그래머스/0/120842. 2차원으로 만들기/2차원으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<int> num_list, int n) 
{
    vector<vector<int>> answer((num_list.size()/n),vector<int> (n,0));
    int i = 0;
    for(int j=0; j<num_list.size()/n; j++)
    {
        for(int k=0; k<n; k++)
        {
            answer[j][k] = num_list[i];
            i++;
        }
    }
    return answer;
}